assets/js/components/DeregistrationModal/DeregistrationModal.test.jsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { faker } from '@faker-js/faker';
+
+import DeregistrationModal from '.';
+
+describe('Deregistration Modal component', () => {
+  it('should render deregistration modal correctly', async () => {
+    const hostName = faker.name.firstName();
+
+    render(
+      <DeregistrationModal
+        hostName={hostName}
+        isOpen
+        onCleanUp={() => {}}
+        onClose={() => {}}
+      />
+    );
+
+    expect(await screen.findByText(new RegExp(hostName))).toBeTruthy();
```

Comment:
This `RegExp` looks out of place. Cannot you simply find the hostname?
There must be tons of examples in our codebase for that

Reply:
This is just some information for anyone wondering:

I was using the `RegExp` to be able to do a _partial_ match on the Host name, as it is part of a bigger string. If you just provide the string like that, then `testing-library` will only look for strings that do an _exact_ match, which doesn't work in this case.

However I found that with the `TextMatch` functions, you can provide an extra argument: `{ exact: false }` to be able to do the partial text match without a `RegExp`.

https://testing-library.com/docs/queries/about/#textmatch

We only do a similar thing in [StatusPill.test.jsx](https://github.com/trento-project/web/blob/main/assets/js/components/HostDetails/StatusPill.test.jsx#L11)